src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -1,43 +1,139 @@
 package cromwell.engine
 
-import akka.actor.Actor
+import akka.actor.{ActorRef, Props, Actor}
 import akka.actor.SupervisorStrategy.Stop
+import akka.event.LoggingReceive
+import cromwell.binding.{FullyQualifiedName, Workflow, WdlBinding, WdlValue}
 
 /**
  * This model assumes workflow actors are transient, created once per workflow instance.
  */
 object WorkflowActor {
 
-  // This is hand-waving over the actual engine-level representation of a workflow.
-  type WorkflowModel = String
+  case class Construct(binding: WdlBinding, inputs: Map[String, WdlValue])
 
-  case class Construct(model: WorkflowModel)
   case object Constructed
 
+  /**
+   * @param badInputs A `Map` of locally qualified input names to a corresponding diagnostic message.
+   */
+  case class ConstructionFailed(badInputs: Map[String, String])
+
   case object Start
+
   case object Started
 
   // Note there is no Stop message defined here, Stop is from Akka.
   case object Stopped
+
+  case class Done(outputs: Map[String, WdlValue])
+
+  case class InvalidOperation(message: String)
+
+  case class Failed(throwable: Throwable)
 }
 
-// Represents the root of a single workflow instance, not a manager of multiple
-// workflows.
+/** Represents the root of a single workflow instance, not a manager of multiple
+  * workflows.
+  */
 class WorkflowActor extends Actor {
+
   import WorkflowActor._
 
+  // This var would be unnecessary in a FSM that #becomes constructed.
+  private var _constructed = false
+
+  private var _executionTable: ExecutionTable = _
+
+  private var _symbolTable: SymbolTable = _
+
+  private var _creator: ActorRef = _
+
+  /**
+   * Check inputs for presence and type compatibility.
+   * TODO break out into a designated "business object", leaving the
+   * TODO actor as a pure boundary layer object?
+   * @param workflow Workflow
+   * @param actualInputs Map of <b>local</b> input names to `WdlValue`s
+   * @return Map of locally qualified input names to a corresponding diagnostic message.
+   */
+  def checkUnsatisfiedInputs(workflow: Workflow, actualInputs: Map[FullyQualifiedName, WdlValue]): Map[FullyQualifiedName, String] =
+    workflow.inputs.collect {
+
+      case requiredInput if !actualInputs.contains(requiredInput._1) =>
+        requiredInput._1 -> "Required workflow input not specified"
+
+      case requiredInput if actualInputs.get(requiredInput._1).get.wdlType != requiredInput._2 =>
+        val expected = actualInputs.get(requiredInput._1).get.wdlType
+        // FIXME formatting
+        requiredInput._1 -> s"Incompatible workflow input types, expected $expected, got ${requiredInput._2}"
+    }
+
   override def receive: Receive = {
+    LoggingReceive {
+      case Construct(binding, actualInputs) if !_constructed =>
+        // TODO Update execution table
+        if (binding == null) {
```

Comment:
Yeah you're right, something should blow up before this if somehow a null bindings is generated.